Mathlib/Data/Matroid/Circuit.lean

```diff
@@ -293,6 +281,35 @@ lemma indep_iff_forall_subset_not_circuit (hI : I ⊆ M.E := by aesop_mat) :
     M.Indep I ↔ ∀ C, C ⊆ I → ¬M.Circuit C := by
   rw [indep_iff_forall_subset_not_circuit', and_iff_left hI]
 
+/-! ### Closure -/
+
+lemma Circuit.closure_diff_singleton_eq (hC : M.Circuit C) (e : α) :
+    M.closure (C \ {e}) = M.closure C :=
+  (em (e ∈ C)).elim
+    (fun he ↦ by rw [(hC.diff_singleton_basis he).closure_eq_closure])
+    (fun he ↦ by rw [diff_singleton_eq_self he])
+
+lemma Circuit.subset_closure_diff_singleton (hC : M.Circuit C) (e : α) :
+    C ⊆ M.closure (C \ {e}) := by
+  rw [hC.closure_diff_singleton_eq]
+  exact M.subset_closure _ hC.subset_ground
+
+lemma Circuit.mem_closure_diff_singleton_of_mem (hC : M.Circuit C) (heC : e ∈ C) :
+    e ∈ M.closure (C \ {e}) :=
+  hC.subset_closure_diff_singleton e heC
+
+lemma exists_circuit_of_mem_closure (he : e ∈ M.closure X) (heX : e ∉ X) :
+    ∃ C ⊆ insert e X, M.Circuit C ∧ e ∈ C :=
+  let ⟨I, hI⟩ := M.exists_basis' X
+  ⟨_, (fundCircuit_subset_insert ..).trans (insert_subset_insert hI.subset),
+    hI.indep.fundCircuit_circuit (by rwa [hI.closure_eq_closure]) (not_mem_subset
+    hI.subset heX), M.mem_fundCircuit e I⟩
+
+lemma mem_closure_iff_exists_circuit_of_not_mem (he : e ∉ X) :
```

Comment:
```suggestion
lemma mem_closure_iff_exists_circuit (he : e ∉ X) :
```
I think this is clear enough. More importantly though, let's keep the name of this and the name of `exists_circuit_of_mem_closure` consistent: either both should have `of_not_mem`, or neither.